### DISCUSSION TITLE
2024-04-10

### DISCUSSION THREAD

**Discussion by yutingzhao1991** [ant-design/ant-design-web3#783] [中文社区双周会]
上周双周会会议记录：https://github.com/ant-design/ant-design-web3/discussions/649

**Comment by yutingzhao1991** [comment]
一些关键内容进度同步：
- ConnectModal 体验优化
- 官网 UI 优化
- 付款组件进展

**Comment by yutingzhao1991** [comment]
SIWE https://docs.ens.domains/web/siwe 支持的需求评审 @jeasonstudio 应该有设计需求，要看看整理下提给 UED

**Comment by kiner-tang** [comment]
要不要搞一个 gas 组件，用于获取和展示 gas 呀，
<img width="404" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/10286961/7f2c50b2-9ae2-43f6-868f-bfed3bfb3443">

**Reply by jeasonstudio** [reply]
觉得这个组件不是很容易做设计上的收敛，基本一个系统一个样，有什么好的想法吗

**Reply by kiner-tang** [reply]
细节待会同步会可以讨论一下

**Reply by kiner-tang** [reply]
感觉可以把一些共性抽离出来，比如说 gas 使用的代币类型、刷新 gas、gas 格式化展示等

**Comment by yutingzhao1991** [comment]
WTF-DApp 高级课程开发计划 https://github.com/WTFAcademy/WTF-Dapp/issues/11

**Reply by yutingzhao1991** [reply]
目前的想法是开发一个 wtfswap，一个简单的，但是完整的去中心化交易所。可以交易以太坊和其他 L2 上面的 ERC20 或者 ERC721 的 token。

教程大纲：
- 概述
  - 什么是去中心化交易所
  - 开发一个去中心化交易所需要什么
- 合约开发
  - 合约设计
  - 合约开发
  - 合约测试和测试环境部署
- 前端部分开发
  - 初始化项目
  - UI 开发
     - 连接钱包开发
     - 交易模块开发
  - 合约调用和调试
  - 支持多链
- 部署上线

UI 类似下面这样，只是整体用 Ant Design + Ant Design Web3 重新设计和重写：

<img width="716" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/1061968/fef1b9a4-acae-4552-b729-cba11b793a71">

**Comment by jeasonstudio** [comment]
几个 issue 讨论：
* https://github.com/ant-design/ant-design-web3/issues/729 这个功能是否要支持
* https://github.com/ant-design/ant-design-web3/issues/745 官网的改版（UED 基本完成了）和移动端适配
* https://github.com/ant-design/ant-design-web3/issues/750 solana 支持更多钱包（扩展讨论下 metamask snap，用 https://tensor.trade/ 演示）
* https://github.com/ant-design/ant-design-web3/issues/739 wallet-connect 配置方式的讨论，包括钱包的分类方式（同构单链、同构多链、异构多链、通用协议）

**Comment by yutingzhao1991** [comment]
- ethers adapter 文档组织方式，现在是和 wagmi 放在一起的，看看要不要分开

**Comment by jeasonstudio** [comment]
bitcoin adapter @AmAzing129
